editor-components/pricing-table/pricing-table7/pricing-table7.tsx

```diff
@@ -2,479 +2,1083 @@ import * as React from "react";
+    const durationIcon = this.getPropValue("icon");
+    const plansDiscountText = this.getPropValue("text2");
+    const subtitle = this.getPropValue("subtitle", { as_string: true });
+    const title = this.getPropValue("title", { as_string: true });
+    const description = this.getPropValue("description", { as_string: true });
+
+    const badgeColors = [
+      'var(--composer-primary-color)',
+      'var(--composer-secondary-color)',
+      'var(--composer-tertiary-color)',
+    ];
+
+    const shouldRenderIconAndDiscount = monthlyText || yearlyText;
+
     return (
       <div className={this.decorateCSS("container")}>
         <div className={this.decorateCSS("max-content")}>
-          {this.castToObject<Pricing[]>("price").map((pricing: any, index: number) => (
-            <div className={this.decorateCSS("card-item-count")} style={{
-              width: 90 / this.getPropValue("itemCount") + "%",
-            }}>
-            <div key={`price7-${index}`} className={this.decorateCSS("price")}>
-              <h2 className={this.decorateCSS("title")}>{pricing.title}</h2>
-              <h1 className={this.decorateCSS("price-text")}>
-                {pricing.price}
-                <p className={this.decorateCSS("duration-text")}>{pricing.duration}</p>
-              </h1>
-              <div className={this.decorateCSS("features")}>
-                {pricing.item.map((data: any, index: number) => {
-                  return (
-                    <li key={`price7-list-${index}`}>
-                      <img src="https://cdn-icons-png.flaticon.com/512/33/33281.png" alt=""/>
-                      {data.itemText}
-                    </li>
-                  );
-                })}
-              </div>
-              <ComposerLink path={pricing.link}>
-                <span className={this.decorateCSS("button")}>
-                  <img src="https://cdn.iconscout.com/icon/free/png-256/whatshot-1782408-1512428.png" alt=""/>
-                  {pricing.buttonText}
-                </span>
-              </ComposerLink>
-            </div>
+          <div className={this.decorateCSS("up-text")}>
+            {subtitle && <span className={this.decorateCSS("subtitle")}>{this.getPropValue("subtitle")}</span>}
+            {title && <span className={this.decorateCSS("title")}>{this.getPropValue("title")}</span>}
+            {description && <div className={this.decorateCSS("description")}>{this.getPropValue("description")}</div>}
+            {shouldRenderIconAndDiscount &&
+              (
+                <div className={this.decorateCSS("duration-items")}>
+                  <>
+                    <span className={this.decorateCSS("text")}>{this.getPropValue("text")}</span>
+                    <div className={this.decorateCSS("switch")} onClick={() => this.setComponentState("plan_type", (this.getComponentState("plan_type") == "monthly-plans") ? "yearly-plans" : "monthly-plans")}>
```

Comment:
Creating and calling a method instead of writing an inline function will increase the readability of the code.